### PR TITLE
modulesync 2.12.0 / zabbix::web: Allow httpd to speak to the database

### PR DESCRIPTION
**Pull request by bastelfreak**
modulesync 2.12.0

### CODE REVIEW

**Comment by bastelfreak** [comment]
so:
* spec/acceptance/server_spec.rb passes locally for me
* I can reduce the 4 errors in spec/acceptance/zabbix_application_spec.rb to 3 with this spatch:

```patch
diff --git a/spec/acceptance/zabbix_application_spec.rb b/spec/acceptance/zabbix_application_spec.rb
index 61129f8..b6d6d65 100644
--- a/spec/acceptance/zabbix_application_spec.rb
+++ b/spec/acceptance/zabbix_application_spec.rb
@@ -38,7 +38,15 @@ describe 'zabbix_application type' do
     # setup zabbix. Apache module isn't idempotent and requires a second run
     it 'works with no error on the first apply' do
       # Cleanup old database
-      shell('/opt/puppetlabs/bin/puppet resource service zabbix-server ensure=stopped; /opt/puppetlabs/bin/puppet resource package zabbix-server-pgsql ensure=purged; rm -f /etc/zabbix/.*done; su - postgres -c "psql -c \'drop database if exists zabbix_server;\'"')
+      cleanup_script = <<-SHELL
+      /opt/puppetlabs/bin/puppet resource service zabbix-server ensure=stopped
+      /opt/puppetlabs/bin/puppet resource package zabbix-server-pgsql ensure=purged
+      rm -f /etc/zabbix/.*done
+      if id postgres > /dev/null 2>&1; then
+        su - postgres -c "psql -c 'drop database if exists zabbix_server;'"
+      fi
+      SHELL
+      shell(cleanup_script)
 
       apply_manifest(pp1, catch_failures: true)
     end

```

(which I would like to refactor into a method so all acceptance test files can use it)

**Comment by bastelfreak** [comment]
I restarted the last successful merge to master build: https://travis-ci.org/github/voxpupuli/puppet-zabbix/jobs/675009991

**Comment by bastelfreak** [comment]
whyyyyyyyyyyyyyyyyyyyyyyyyyyy does it pass

**Comment by bastelfreak** [comment]
I created a simple PR against master that only touches our README.md: https://github.com/voxpupuli/puppet-zabbix/pull/681

**Comment by bastelfreak** [comment]
which passed. Now I cherry-picked the cleanup commit into another PR:
https://github.com/voxpupuli/puppet-zabbix/pull/682

**Comment by bastelfreak** [comment]
@baurmatt I'm still debugging this, but if you've any ideas why this works on master but fails here, please let me know :)

**Comment by bastelfreak** [comment]
I've no clue why the code in docker fails to create a database for one zabbix version:
https://travis-ci.org/github/voxpupuli/puppet-zabbix/jobs/679220993#L9204


I'm unable to reproduce this locally with vagrant

**Comment by baurmatt** [comment]
@bastelfreak Ha! I've seen something similar with Ubuntu 18.04. Turned out that it ships a `/etc/dpkg/dpkg.cfg.d/excludes` which includes the following directories by default:

```
# Drop all man pages
path-exclude=/usr/share/man/*

# Drop all documentation ...
path-exclude=/usr/share/doc/*

# ... except copyright files ...
path-include=/usr/share/doc/*/copyright

# ... and Debian changelogs
path-include=/usr/share/doc/*/changelog.Debian.*
```

This led to /usr/share/doc/zabbix-*-pgsql-3.4*/*.sql not being installed and hence failing tests.

I think the problem here is, that `nodocs` isn't removed anymore from /etc/yum.conf. That's probably the equivalent in CentOS.

File: spec/spec_helper_acceptance.rb [hunk 1]
```diff
@@ -1,21 +1,5 @@
...
-install_ca_certs unless ENV['PUPPET_INSTALL_TYPE'] =~ %r{pe}i
-install_module_on(hosts)
-install_module_dependencies_on(hosts)
-
-RSpec.configure do |c|
-  # Readable test descriptions
-  c.formatter = :documentation
-  hosts.each do |host|
-    if host[:platform] =~ %r{el-7-x86_64} && host[:hypervisor] =~ %r{docker}
-      on(host, "sed -i '/nodocs/d' /etc/yum.conf")
```
Comment (baurmatt) [inline spec/spec_helper_acceptance.rb:16]: @bastelfreak This shitty thing! :D

Comment (ekohl) [reply spec/spec_helper_acceptance.rb:16]: So we need this? What breaks?

Comment (baurmatt) [reply spec/spec_helper_acceptance.rb:16]: I leads to files in /usr/share/doc/* not being installed. As Zabbix places their database initialization script there, this breaks our acceptance CI.

Comment (ekohl) [reply spec/spec_helper_acceptance.rb:16]: Ah yes, I now see the IRC conversation. Pointing @bastelfreak in the right direction.

Comment (bastelfreak) [reply spec/spec_helper_acceptance.rb:16]: Thanks @baurmatt . We figured the same out in parallel yesterday on IRC. The current code should now work again on docker images. Are you already working on Ubuntu 18/20 support for the module?

**Pull request merged**